### PR TITLE
Updated callbacks.py

### LINKED ISSUE #17305: Early stopping callback will stop early even if improvement when patience = 0
keras==2.7.0

The early stopping callback source code https://github.com/keras-team/keras/blob/v2.11.0/keras/callbacks.py#L1942-L2114 has the functionality where self.wait is set to zero if improvement occurs (L2080), but then in L 2083 has a check for stopping if `self.wait >= self.patience`, meaning (due to the greater than OR equal in L2093) the training will always stop early if patience = 0, even if the monitored metric is improving. 

This conflicts with the description of patience as the minimum number of epochs to wait if improvement does *not* occur.  It seems that L2083 should have a > operator and not >=.

### PR DESCRIPTION
**Pull request by sushreebarsa**
Added an operator in the L2082 in TF doc as it was raising conflict for early stopping callback, if the monitored metric is improving. Fixes #17305

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -2079,7 +2079,7 @@ def on_epoch_end(self, epoch, logs=None):
                 self.wait = 0
 
         # Only check after the first epoch.
-        if self.wait >= self.patience and epoch > 0:
+        if self.wait > self.patience and epoch > 0:
```
[the remark this answers is not in the document]
Comment (rchao) [reply keras/callbacks.py:2082]: Alternatively, add a return after L2079. Will this work?

**Comment by old-school-kid** [comment]
@rchao I guess this can be closed since #17366 is a better approach.

**Comment by rchao** [comment]
> @rchao I guess this can be closed since #17366 is a better approach.

Yes, thus closing this. Please reopen if you feel that doesn't fix the original issue. Thanks.
